example/app/(examples)/DataBinding.tsx

```diff
@@ -1,23 +1,48 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import {
   SafeAreaView,
   ScrollView,
   StyleSheet,
   Button,
   View,
 } from 'react-native';
-import Rive, { Fit, RiveRef } from 'rive-react-native';
+import Rive, {
+  Fit,
+  RiveRef,
+  useRiveColor,
+  useRiveNumber,
+  useRiveReady,
+  useRiveString,
+} from 'rive-react-native';
 
 export default function DataBinding() {
   const riveRef = React.useRef<RiveRef>(null);
+  const riveIsReady = useRiveReady(riveRef);
 
+  let [buttonText, setButtonText] = useRiveString(riveRef, 'Button/State_1');
+  let [lives, setLives] = useRiveNumber(riveRef, 'Energy_Bar/Lives');
+  let [barColor, setBarColor] = useRiveColor(riveRef, 'Energy_Bar/Bar_Color');
+
+  useEffect(() => {
+    if (riveIsReady) {
+      // Set initial values through hooks
+      setButtonText("Let's go!");
```

Comment:
It looks like this is getting fired before `setButtonText` and the others are defined. Changing the if statement to this fixes it: 

```
if (riveIsReady && setButtonText && setLives && setBarColor) {
```

Reply:
If you're testing on iOS you may not see the correct values updated as the iOS implementation isn't done. There the setters are always defined, but they will log out a DEV warning if the ref isn't available for whatever reason. Are you seeing that warning?